bump_pydantic/main.py

```diff
@@ -59,6 +67,14 @@ def main(
     """
     console = Console(log_time=True)
     console.log("Start bump-pydantic.")
+
+    if platform.system() == "Windows" and number_of_jobs > _MAX_JOBS_SUPPORTED_DEFAULT:
+        console.log(
+            f"Given number of jobs {number_of_jobs} exceeds maximum allowed number "
+            f"{_MAX_JOBS_SUPPORTED_DEFAULT}. Using {_MAX_JOBS_SUPPORTED_DEFAULT} instead."
+        )
+        number_of_jobs = _MAX_JOBS_SUPPORTED_DEFAULT
```

Comment:
Do we need to check the OS? Can't we just get the min between the two?

Reply:
My understanding is that the behavior is only required on Windows.  Hence on Linux I could use more jobs than I have logical cores...